28 only fourth image uploads

straiforos opened the pull request:
- Fixed embedded media array being reinstantiated on each image upload causing only the last image to be linked to the post.
- Consolidated env configuration into AppConfig class due to additional test mode added for easier integration testing outside of unit tests.
- Min-Max date unit tests added.

marcomaroni-github commented:
@straiforos I added height and width dimensions to the images so the gray bada no longer appears, check the code and do some tests, if ok add or update the unit tests, then I proceed with the merge and release. Thanks.

straiforos commented:
Thanks @marcomaroni-github,

I noticed the breakout of classes up I did in #21 was incomplete. I did not clean up the classes in the main media.ts file. I am making sure we only have that aspect ratio change in one place and adding unit tests to it.

straiforos commented:
@marcomaroni-github I added unit tests and updated the mocks for existing ones to pass. 

I will create an additional PR to do the clean up that won't change functionality but remove the need for the duplication you had to do to make sure the aspect ratio was in the `src/media/media.ts` and the `src/media/processors/InstagramImageProcessor.ts`.

Ran both the image and images test modes and are looking good! 

You can merge it when you can :)
